frontend/public/components/resource-quota.jsx

```diff
@@ -13,6 +13,100 @@ const menuActions = [...common];
 
 const quotaKind = quota => quota.metadata.namespace ? referenceForModel(ResourceQuotaModel) : referenceForModel(ClusterResourceQuotaModel);
 
+const humanizeQuotaResource = (resourceType, useTitleCase) => {
+  if (!resourceType) {
+    return resourceType;
+  }
+
+  const nameTitleCaseFormatMap = {
+    'configmaps': 'Config Maps',
+    'cpu': 'CPU (Request)',
+    'limits.cpu': 'CPU (Limit)',
+    'limits.memory': 'Memory (Limit)',
+    'memory': 'Memory (Request)',
+    'openshift.io/imagesize': 'Image Size',
+    'openshift.io/imagestreamsize': 'Image Stream Size',
+    'openshift.io/projectimagessize': 'Project Image Size',
+    'persistentvolumeclaims': 'Persistent Volume Claims',
+    'requests.storage': 'Storage (Request)',
+    'pods': 'Pods',
+    'replicationcontrollers': 'Replication Controllers',
+    'requests.cpu': 'CPU (Request)',
+    'requests.memory': 'Memory (Request)',
+    'resourcequotas': 'Resource Quotas',
+    'secrets': 'Secrets',
+    'services': 'Services',
+    'services.loadbalancers': 'Service Load Balancers',
+    'services.nodeports': 'Service Node Ports',
+  };
+
+  const nameFormatMap = {
+    'configmaps': 'config maps',
+    'cpu': 'CPU (request)',
+    'limits.cpu': 'CPU (limit)',
+    'limits.memory': 'memory (limit)',
+    'memory': 'memory (request)',
+    'openshift.io/imagesize': 'image size',
+    'openshift.io/imagestreamsize': 'image stream size',
+    'openshift.io/projectimagessize': 'project image size',
+    'persistentvolumeclaims': 'persistent volume claims',
+    'requests.storage': 'storage (request)',
+    'replicationcontrollers': 'replication controllers',
+    'requests.cpu': 'CPU (request)',
+    'requests.memory': 'memory (request)',
+    'resourcequotas': 'resource quotas',
+    'services.loadbalancers': 'service load balancers',
+    'services.nodeports': 'service node ports',
+  };
+  if (useTitleCase) {
+    return nameTitleCaseFormatMap[resourceType] || resourceType;
+  }
+  return nameFormatMap[resourceType] || resourceType;
+};
+
+const compareResourceType = (left, right) => {
+  // CPU Request
+  if (left === 'cpu' || left === 'requests.cpu') {
+    return right === 'cpu' || right === 'requests.cpu' ? 0 : -1;
+  }
+  if (right === 'cpu' || right === 'requests.cpu') {
+    return 1;
+  }
+
+  // Memory Request
+  if (left === 'memory' || left === 'requests.memory') {
```

Comment:
I'd suggest we order it like

CPU (requests)
CPU (limits)
Memory (requests)
Memory (limits)